src/tests/tracing/userevents/custommetadata/custommetadata.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Threading;
+using Tracing.UserEvents.Tests.Common;
+using Microsoft.Diagnostics.Tracing;
+
+namespace Tracing.UserEvents.Tests.CustomMetadata
+{
+    public static class CustomMetadata
+    {
+        public static void CustomMetadataTracee()
+        {
+            // Allow some time for tracepoints be enabled before emitting events
+            Thread.Sleep(150);
```

Comment:
Maybe you should put this logic in a helper used by all tests? Like putting all this logic in the test runner and then only do call to Run from each tests Main method:

```
public static int Main(string[] args)
{
    return UserEventsTestRunner.Run(args, "custommetadata", typeof(CustomMetadata).Assembly.Location, s_eventGeneratingAction, s_traceValidator, 20000);
}
```

and then you can put the wait time and other coordinating activities as part of Run method and all tests will work the same.